pkg/lib/cockpit.js

```diff
@@ -2718,16 +2703,38 @@ function factory() {
      * Dark mode
      */
     (function() {
-        if (window.matchMedia('(prefers-color-scheme: dark)').matches) {
-            cockpit.dark_mode = true;
-        } else {
-            cockpit.dark_mode = false;
-        }
+        const setTheme = theme => {
+              if ((window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches && theme === "auto") || theme === "dark") {
+                  document.documentElement.classList.add('pf-theme-dark');
+                  cockpit.dark_mode = true;
+              } else {
+                  document.documentElement.classList.remove('pf-theme-dark');
+                  cockpit.dark_mode = false;
+              }
+        };
+
+        const theme = localStorage.getItem('shell:mode') || 'auto';
+        setTheme(theme);
+
+        window.onstorage = event => {
+            if (event.key !== "shell:mode")
+                return;
+
+            setTheme(event.newValue);
+        };
 
         window.matchMedia('(prefers-color-scheme: dark)').addEventListener('change', event => {
+            const theme = localStorage.getItem('shell:mode') || 'auto';
+            // If a user set's an explicit theme, ignore system changes.
+            if (theme !== "auto")
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17914-20221115-124418-c7be4864-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)